### PR TITLE
[ws-manager-bridge] Also copy status.conditions.headlessTaskFailed into WorkspaceInstanceConditions

### PR DESCRIPTION
**Pull request by jankeromnes**
## Description

In the new Teams & Projects UI, we've introduced a Prebuild Status widget.

To determine a Prebuild's status, depending on the context of the page, the widget sometimes relies on:
1. A [prebuild.{state,error}](https://github.com/gitpod-io/gitpod/blob/afd2cb3a2a7da004412fd45742609daa16019105/components/gitpod-protocol/src/protocol.ts#L634-L645) (e.g. when listing Prebuilds for a Project/Branch)
2. A [instance.{phase,conditions}](https://github.com/gitpod-io/gitpod/blob/afd2cb3a2a7da004412fd45742609daa16019105/components/gitpod-protocol/src/workspace-instance.ts#L56-L87) (e.g. when showing the live logs of a Prebuild)

If an `init` task fails during a Prebuild, this creates a `status.conditions.headlessTaskFailed` error.
1. If there is such an error, it [gets copied into](https://github.com/gitpod-io/gitpod/blob/46feeac4c9d5ace28a9a565ca9d81b8a2f4e09d3/components/ws-manager-bridge/ee/src/bridge.ts#L81-L83) the `prebuild.error` -- so the first widget implementation can support & show this error state
2. However, it [does not currently get copied](https://github.com/gitpod-io/gitpod/blob/b031cf3296931f383c60c47bcb6fa53557861bd1/components/ws-manager-bridge/src/bridge.ts#L162-L167) into the `instance.conditions` -- so the second widget implementation is not aware of such errors, and cannot show them

Proposed solution:
- Copy `status.conditions.headlessTaskFailed` into the `instance.conditions` (so that the instance-based status widget can watch out for such errors)

## How to test

1. Trigger a prebuild with an init task like `- init: oops`
2. The prebuild workspace instance should have a `instance.conditions.headlessTaskFailed` when the above fails

## Release Notes

```release-note
NONE
```

/uncc

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

Approval requirements bypassed by manually added approval.

This pull-request has been approved by:

*No associated issue*. Requirement bypassed by manually added approval.

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- **[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)**
- **[components/ws-manager-bridge/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-manager-bridge/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

Approval requirements bypassed by manually added approval.

This pull-request has been approved by:

*No associated issue*. Requirement bypassed by manually added approval.

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- **[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)**
- **[components/ws-manager-bridge/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-manager-bridge/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by svenefftinge — APPROVED** [review]
LGTM

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: eddb2671bd28642c31b82ed5071e49e26ffab05f</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

Approval requirements bypassed by manually added approval.

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/5885#pullrequestreview-765016406" title="Approved">svenefftinge</a>*

*No associated issue*. Requirement bypassed by manually added approval.

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)~~ [svenefftinge]
- ~~[components/ws-manager-bridge/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/ws-manager-bridge/OWNERS)~~ [svenefftinge]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
